fvtest/compilertriltest/MaxMinTest.cpp

```diff
@@ -159,3 +185,139 @@ INSTANTIATE_TEST_CASE_P(MaxMin, Int64MaxMin, ::testing::Combine(
+        << "The " << arch << " code generator currently doesn't support fmax (see issue #4176)";
+    auto param = TRTest::to_struct(GetParam());
+    std::cout << "Running test on: " << arch << std::endl;
+    char inputTrees[1024] = {0};
+    std::snprintf(inputTrees, sizeof(inputTrees),
+        "(method return=Float"
+        "  (block"
+        "    (freturn"
+        "      (%s"
+        "        (fconst %f )"
+        "        (fconst %f ))"
+        ")))",
+        param.opcode.c_str(),
+        param.lhs,
+        param.rhs);
+    auto trees = parseString(inputTrees);
+
+    ASSERT_NOTNULL(trees);
+
+    Tril::DefaultCompiler compiler(trees);
+
+    ASSERT_EQ(0, compiler.compile()) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
+
+    auto entry_point = compiler.getEntryPoint<float_t (*)(void)>();
+    ASSERT_EQ(param.oracle(param.lhs, param.rhs), entry_point());
+}
+
+TEST_P(FloatMaxMin, UsingLoadParam) {
+    std::string arch = omrsysinfo_get_CPU_architecture();
+    SKIP_IF(OMRPORT_ARCH_X86 == arch || OMRPORT_ARCH_S390 == arch || OMRPORT_ARCH_S390X == arch || OMRPORT_ARCH_HAMMER == arch || OMRPORT_ARCH_AARCH64 == arch, KnownBug)
+        << "The " << arch << " code generator currently doesn't support fmax (see issue #4176)";
+    auto param = TRTest::to_struct(GetParam());
+
+    char inputTrees[1024] = {0};
+    std::snprintf(inputTrees, sizeof(inputTrees),
+        "(method return=Float args=[Float, Float]"
+        "  (block"
+        "    (freturn"
+        "      (%s"
+        "        (fload parm=0)"
+        "        (fload parm=1))"
+        ")))",
+        param.opcode.c_str());
+    auto trees = parseString(inputTrees);
+
+    ASSERT_NOTNULL(trees);
+
+    Tril::DefaultCompiler compiler(trees);
+
+    ASSERT_EQ(0, compiler.compile()) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
+
+    auto entry_point = compiler.getEntryPoint<float_t (*)(float_t, float_t)>();
+    ASSERT_EQ(param.oracle(param.lhs, param.rhs), entry_point(param.lhs, param.rhs));
+}
+
+
+INSTANTIATE_TEST_CASE_P(MaxMin, FloatMaxMin, ::testing::Combine(
+    ::testing::ValuesIn(
+    TRTest::filter(TRTest::const_value_pairs<float, float>(), smallFp_filter<float>)),
+    ::testing::Values(
+        std::make_tuple<const char*, float_t(*)(float_t, float_t)>("fmax", f_max),
+        std::make_tuple<const char*, float_t(*)(float_t, float_t)>("fmin", f_min))));
+
+class DoubleMaxMin : public TRTest::BinaryOpTest<double_t> {};
+
+TEST_P(DoubleMaxMin, UsingConst) {
+    std::string arch = omrsysinfo_get_CPU_architecture();
+    SKIP_IF(OMRPORT_ARCH_X86 == arch || OMRPORT_ARCH_S390 == arch || OMRPORT_ARCH_S390X == arch || OMRPORT_ARCH_HAMMER == arch || OMRPORT_ARCH_AARCH64 == arch, KnownBug)
+        << "The " << arch << " code generator currently doesn't support fmax (see issue #4176)";
```

Comment:
Technically, the opcodes here are dmax and dmin.
```suggestion
        << "The " << arch << " code generator currently doesn't support dmax/dmin (see issue #4176)";
```